orc/cmd/resource-generator/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	_ "embed"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+const (
+	defaultYear       = "2024"
+	defaultAPIVersion = "v1alpha1"
+)
+
+//go:embed data/api.template
+var api_template string
+
+type specExtraValidation struct {
+	Rule    string
+	Message string
+}
+
+type templateFields struct {
+	APIVersion           string
+	Year                 string
+	Name                 string
+	StatusExtraType      string
+	SpecExtraValidations []specExtraValidation
+}
+
+var allResources []templateFields = []templateFields{
+	{
+		Name:       "Image",
+		APIVersion: "v1alpha1",
+		SpecExtraValidations: []specExtraValidation{
+			{
+				Rule:    "!has(self.__import__) ? has(self.resource.content) : true",
```

Comment:
This is a good first step, but going forward we probably want a more structured way to define required fields in the spec...

Reply:
This particular one is special because the Image controller is a snowflake. This is a controller-specific validation at the *spec* level. I don't expect this to be a common pattern.